app/util/appUtil.js

```diff
@@ -47,35 +48,58 @@ api.on('error', (err) => {
 
 async function addFile(file) {
   const form = new FormData()
-  form.append('file', fs.createReadStream(file))
-  const body = await fetch(`http://${IPFS_HOST}:${IPFS_PORT}/api/v0/add?cid-version=0`, {
+  form.append('file', fs.createReadStream(file.path), file.name)
+  const body = await fetch(`http://${IPFS_HOST}:${IPFS_PORT}/api/v0/add?cid-version=0&wrap-with-directory=true`, {
     method: 'POST',
     body: form,
   })
-  return body.json()
+
+  // Build string of objects into array
+  const text = await body.text()
+  const json = text
+    .split('\n')
+    .filter((obj) => obj.length > 0)
+    .map((obj) => JSON.parse(obj))
+
+  return json
 }
 
 async function processMetadata(file) {
   if (file) {
-    const response = await addFile(file)
-    if (response && response.Hash && response.Name && response.Size) {
-      const decoded = bs58.decode(response.Hash)
+    const responses = await addFile(file)
+
+    // directory has no Name
+    const dir = responses.find((r) => r.Name === '')
+    if (dir && dir.Hash && dir.Size) {
+      const decoded = bs58.decode(dir.Hash)
       return `0x${decoded.toString('hex').slice(4)}`
     }
   }
 
   return null
 }
 
-const downloadFile = async (hash) => {
-  const url = `http://${IPFS_HOST}:${IPFS_PORT}/api/v0/cat?arg=${hash}`
-  const res = await fetch(url, { method: 'POST' })
-
-  if (!res.ok) {
-    throw new Error(`Error fetching file from IPFS (${res.status}): ${await res.text()}`)
-  }
-
-  return res.body
+const downloadFile = async (dirHash) => {
```

Comment:
Hang on does this deal with the old case where we just store the file and not the directory

Reply:
Not yet!